js/formDataToJson.js

```diff
@@ -91,6 +95,206 @@ async function copyToClipboard(event){
+{
+	const {owner, repo} = getOrgAndRepoArgsGitHub(projectURL);
+	const FILE_PATH = 'code.json'
+	const createFileApiUrl = `https://api.github.com/repos/${owner}/${repo}/contents/${FILE_PATH}`;
+	const encodedContent = btoa(codeJSONObj);
+	console.log("Content: ", encodedContent);
+	console.log("Branch: ", NEW_BRANCH);
+
+	const response = await fetch(createFileApiUrl, 
+		{
+			method: 'PUT',
+			headers: {
+				'Accept': 'application/vnd.github+json',
+				'Authorization': 'Bearer '.concat(token),
+				'X-GitHub-Api-Version': "2022-11-28"
+			},
+			body: JSON.stringify({
+				message: "Add codejson to project",
+				committer: {
+					name: "CodeJSON formsite",
+					email: "opensource@cms.hhs.gov"
+				},
+				content: encodedContent,
+				branch: NEW_BRANCH,
+			}),
+		}
+	);
+
+	const data = await response.json()
+
+	if ( response.ok )
+	{
+		console.log('File added successfully: ', data);
+		return true;
+	}
+	else
+	{
+		console.error('Error adding file: ', data);
+		return false;
+	}
+}
+
+async function createPR(projectURL, token)
+{
+	const {owner, repo} = getOrgAndRepoArgsGitHub(projectURL);
+	const createPrApiUrl = `https://api.github.com/repos/${owner}/${repo}/pulls`;
+	const response = await fetch(createPrApiUrl, 
+		{
+			method: 'POST',
+			headers: {
+				'Content-Type': 'application/json',
+				'Authorization': 'token '.concat(token),
+				'X-GitHub-Api-Version': "2022-11-28"
+			},
+			body: JSON.stringify({
+				title: "Add code.json to Project",
+				body: "Add generated code.json file to project. Code.json was generated via a form.io form site.",
+				head: NEW_BRANCH,
+				base: 'main',
+
+			}),
+		}
+	);
+
+	const data = await response.json();
+
+	if (response.ok)
+	{
+		console.log('Pull request created successfully: ', data);
+		return true;
+	}
+	else
+	{
+		console.error("Error creating PR!: ", data);
+		return false;
+	}
+}
+
+// Creates PR on requested project
+async function createProjectPR(event){
+	event.preventDefault();
+
+	var textArea = document.getElementById("json-result");//Step 1
+	var codeJSONObj = JSON.parse(textArea.value)
+	
+	if('gh_api_key' in window)
+	{
+		var apiKey = window.gh_api_key;
+		
+		if ('repositoryURL' in codeJSONObj)
+		{
+			var prCreated = false;
+			//Step 1
+			const branchCreated = await createBranchOnProject(codeJSONObj.repositoryURL,apiKey);
+			if (branchCreated)
+			{
+				const fileAdded = await addFileToBranch(codeJSONObj.repositoryURL, apiKey, textArea.value);
+
+				if (fileAdded)
+				{
+					prCreated = await createPR(codeJSONObj.repositoryURL, apiKey);
+					if(prCreated)
+					{
+						console.log("PR successfully created!");
+						alert("PR has been created!");
+					}
+				}
+			}
+		}
+		else
+		{
+			console.error("No URL found!");
+			alert("No URL given for project!");
+		}
+		
+	}
+	else
+	{
+		console.error("No API key found!");
+		alert("No Api Key in submitted data!");
+	}
+	//console.log(codeJSONObj)
+}
```

Comment:
In this main function here, could we add an alert for cases where there was an error with the GitHub API? User could've provided a projectURL and a working key but GitHub API could've returned an error

Reply:
here's an example I came across, it would be helpful to communicate this to the user if they didn't put in the key with the correct permissions and add an alert statement with how they can fix
<img width="1077" alt="Screenshot 2025-04-09 at 3 22 00 PM" src="https://github.com/user-attachments/assets/6f49fc8c-d53a-428b-8790-149b482dc334" />

Reply:
Sounds good to me I added some alerts that give the status of the api requests if they fail.